lib/irb/cmd/help.rb

```diff
@@ -1,55 +1,22 @@
-# frozen_string_literal: false
-#
-#   help.rb - helper using ri
-#
+# frozen_string_literal: true
 
-require_relative "nop"
+require_relative "show_doc"
 
 module IRB
-  # :stopdoc:
-
   module ExtendCommand
-    class Help < Nop
-      class << self
-        def transform_args(args)
-          # Return a string literal as is for backward compatibility
-          if args.empty? || string_literal?(args)
-            args
-          else # Otherwise, consider the input as a String for convenience
-            args.strip.dump
-          end
-        end
-      end
-
+    class Help < ShowDoc
       category "Context"
-      description "Enter the mode to look up RI documents."
-
-      def execute(*names)
-        require 'rdoc/ri/driver'
-
-        unless self.class.const_defined?(:Ri)
-          opts = RDoc::RI::Driver.process_args([])
-          self.class.const_set(:Ri, RDoc::RI::Driver.new(opts))
-        end
+      description "[DEPRECATED] Enter the mode to look up RI documents."
 
-        if names.empty?
-          Ri.interactive
-        else
-          names.each do |name|
-            begin
-              Ri.display_name(name.to_s)
-            rescue RDoc::RI::Error
-              puts $!.message
-            end
-          end
-        end
+      DEPRECATION_MESSAGE = <<~MSG
+        [Deprecation] The `help` command will be repurposed to show command help in the future.
```

Comment:
Looks good!
Just an idea, I think it would be great if there's a message to indicate that the command `to show command help` is `show_cmds`?